### ISSUE TITLE
Introduce drush self-update

### ISSUE THREAD

**Issue by DuaelFr** [drush-ops/drush#1980]
As composer does, it would be great to allow drush to update itself according to the way it has been installed (via composer locally, via composer globally or via the phar file).
Thanks :)

**Comment by xyfantis** [comment]
:+1:

**Comment by weitzman** [comment]
Makes sense for phar users. See https://github.com/padraic/phar-updater and his implementation at https://github.com/padraic/humbug/blob/master/src/Command/SelfUpdate.php. 

Composer users should just use composer.
